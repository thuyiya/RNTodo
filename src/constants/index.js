const theme = {
    colors: {
        primary: '#0145a6',
        textColor: '#333333',
        seconderyTextColor: '#787878',
        backdropColor: '#000000',
        loadingIndicator: '#000000',
        background: '#f0f0f0',
        disbaleText: '#c9c9c9',
        white: '#ffffff',
        buttonText: '#ffffff',
    },
    opacity: {
        backdrop: 0.3
    },
    fontSizeMap: {
        h1: 24,
        h2: 22,
        h3: 20,
        h4: 18,
        h5: 16,
        h6: 14
    },
    fontWeightMap: {
        normal: '300',
        avarage: '400',
        lightBold: '500',
        midBold: '700',
        bold: '900',
    },
    spaceSize: 8,
}

const strings = {
    AUTHENTICATION_SCREEN: {
        DEVICE_CAPABILITY: 'No face or fingerprint scanner is available on the device.',
        PROCCED_TEXT: 'Set Authentication to Proceed',
        GO_TO_SETTINGS: 'Go to Settings',
        PROCEED_TO_AUTH: 'Proceed',
        GO_TO_SETTINGS_ALERT_TITLE: 'Authentication Setup',
        GO_TO_SETTINGS_ALERT_DESCRIPTION: 'Go to settings and setup authentication to proceed',
        APP_ACCESS: 'Access setup',
        APP_ACCESS_DESCRIPTION: 'You have to pass your authentication process to see todos',
    },
    HOME: {
        TITLE: 'TODO: '
    },
    EMPTYVIEW: {
        NO_DATA: "No Todo's Found"
    }
}

const actionTypes = {
    TODO: {
        ADD_ITEM: 'TODO_ADD_ITEM',
        CLEAN_ITEMS: 'CLEAN_ITEMS',
        REMOVE_ITEMS: 'REMOVE_ITEMS',
        SET_SELECTED_ITEM: 'SET_SELECTED_ITEM'
    }
}

export { theme, strings, actionTypes }