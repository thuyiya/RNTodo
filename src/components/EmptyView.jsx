import React from "react";
import { View, StyleSheet, Text, Dimensions} from "react-native";
import { theme, strings } from "../constants";
import Typography from "./Typography";

const { width, height } = Dimensions.get('screen')

const EmptryView = () => {
  return (
    <View style={styles.container}>
      <Typography color={theme.colors.disbaleText} >{strings.EMPTYVIEW.NO_DATA}</Typography>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: width,
    height: height *0.6,
    justifyContent: "center",
    alignItems: "center",
  }
});

export default EmptryView;
