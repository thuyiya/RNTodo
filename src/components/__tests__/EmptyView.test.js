import React from "react";
import renderer from "react-test-renderer";
import {render, screen} from '@testing-library/react-native';
import EmptyView from "../EmptyView";
import { strings } from "../../constants";

describe("<EmptyView />", () => {
  it("has 1 children", () => {
    const tree = renderer.create(<EmptyView />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  it(`displays the ${strings.EMPTYVIEW.NO_DATA}`, () => {
    render(<EmptyView />);
    expect(screen.queryByText(strings.EMPTYVIEW.NO_DATA)).toBeTruthy();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<EmptyView />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
