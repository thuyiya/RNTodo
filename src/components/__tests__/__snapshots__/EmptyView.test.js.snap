// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyView /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "height": 800.4,
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#c9c9c9",
          "fontSize": 16,
          "fontWeight": "300",
          "margin": 0,
          "textAlign": "center",
        },
        {},
      ]
    }
    testID="text"
  >
    No Todo's Found
  </Text>
</View>
`;
